documentation/decisions/adr/2023-12-15-deliverable-reporting-strategy.md

```diff
@@ -0,0 +1,235 @@
+  - Supports grouping both within the GitHub projects and within GitHub insight reporting.
+- **Cons**
+  - Prevents the engineering team from using GitHub milestones to organize and group related issues outside of the 30k deliverable framework.
+  - Milestones that represent 30k deliverables will have a *lot* of tickets unless we work on downscoping 30k deliverables.
+  - Does not support customizing the order of the groups on the sprint board when grouping by milestone -- milestones can only be sorted alphabetically.
+
+### Label
+
+This option involves creating a label for each 30k deliverable with a consistent prefix (e.g. `30k: Public launch`) and then applying this label to each of the issues required for delivery of that 30k.
+
+> [!TIP]
+> **Bottom line:** This is the best option if:
+> - we want a consistent way to filter issues by deliverable in the repository and across GitHub projects
+> - but can compromise on being able to **group** by deliverable in GitHub projects or reporting
+
+- **Pros**
+  - Allows the engineering team to continue to use GitHub milestones to organize issues into units of work that make sense to them.
+  - Supports filtering both within the repository and within GitHub projects (e.g. roadmap and sprint board).
+- **Cons**
+  - Does not easily support grouping in GitHub projects or GitHub insights reporting.
+  - Will result in a large number of labels if 30k labels aren't deleted once a 30k deliverable is complete.
+
+### Deliverable column
+
+This option involves creating a single select "deliverable" column in the GitHub projects, with values for each 30k deliverable, and then selecting the corresponding "deliverable" value for each issue added to the project.
+
+> [!TIP]
+> **Bottom line:** This is the best option if:
+> - we want a consistent way of filtering, grouping, and sorting issues by deliverable in GitHub projects and reporting
+> - but can compromise on being able to filter by deliverable in the **repository**
+
+- **Pros**
+  - Allows the engineering team to continue to use GitHub milestones to organize issues into units of work that make sense to them.
+  - Supports filtering within GitHub projects (e.g. roadmap and sprint board).
+  - Supports grouping within GitHub projects and within GitHub insight reporting.
+  - Supports customizing the order of groups on the sprint board when grouping by deliverable -- for example we can make "Static site public launch" appear above "GET Opportunities" even though GET opportunities would be first alphabetically.
+- **Cons**
+  - Does not support filtering or searching within the GitHub repository.
+  - Will result in a large number of values for the deliverable column if not regularly maintained.
+  - List of deliverable column values will have to be maintained separately across projects -- inconsistencies between these lists may introduce bugs in custom reporting.
```

Comment:
Maybe there's a linter we can build to help check for consistency?

Reply:
Added a note about linting in this commit: [fix: Incorporates PR comments](https://github.com/HHS/simpler-grants-gov/pull/926/commits/5831bfabf360fca818dc281ddd31129b6152a7bd)